YSI_Extra/y_inline_mselection.inc

```diff
@@ -0,0 +1,173 @@
+	Masterchen09
+	Southclaws
+	PatchwerkQWER
+	m0k1
+	paulommu
+	udan111
+
+Thanks:
+	JoeBullet/Google63 - Handy arbitrary ASM jump code using SCTRL.
+	ZeeX - Very productive conversations.
+	koolk - IsPlayerinAreaEx code.
+	TheAlpha - Danish translation.
+	breadfish - German translation.
+	Fireburn - Dutch translation.
+	yom - French translation.
+	50p - Polish translation.
+	Zamaroht - Spanish translation.
+	Los - Portuguese translation.
+	Dracoblue, sintax, mabako, Xtreme, other coders - Producing other modes for
+		me to strive to better.
+	Pixels^ - Running XScripters where the idea was born.
+	Matite - Pestering me to release it and using it.
+
+Very special thanks to:
+	Thiadmer - PAWN, whose limits continue to amaze me!
+	Kye/Kalcor - SA:MP.
+	SA:MP Team past, present and future - SA:MP.
+
+Optional plugins:
+	Gamer_Z - GPS.
+	Incognito - Streamer.
+	Me - sscanf2, fixes2, Whirlpool.
+*/
+
+#if !defined _INC_y_inline
+	#tryinclude "..\YSI_Coding\y_inline\y_inline_entry"
+#endif
+#if !defined _INC_y_inline
+	#tryinclude "YSI_Coding\y_inline\y_inline_entry"
+#endif
+#if !defined _INC_y_inline
+	#tryinclude "YSI\YSI_Coding\y_inline\y_inline_entry"
+#endif
+#if !defined _INC_y_inline
+	#tryinclude <YSI_Coding\y_inline\y_inline_entry>
+#endif
+#if !defined _INC_y_inline
+	#tryinclude <YSI\YSI_Coding\y_inline\y_inline_entry>
+#endif
+#if !defined _INC_y_inline
+	#error Could not find y_inline
+#endif
+
+stock const
+	Func:YSI_g_sRequestModelSelection[MAX_PLAYERS]<ii>,
+	YSI_g_PlayerCurrentListID[MAX_PLAYERS] = {mS_INVALID_LISTID, ...};
+
+stock ShowModelSelectionInline(Func:cb<ii>, playerid, listid, const headerText[], dialogBGcolor = 0x4A5A6BBB, previewBGcolor = 0x88888899, selectionColour = 0xFFFF00AA) 
+{
+	new ret = ShowModelSelectionMenu(playerid, listid, headerText, dialogBGcolor, previewBGcolor, selectionColour);
+
+	if (ret) 
+	{
+		Indirect_Claim(cb);
+		YSI_g_PlayerCurrentListID[playerid] = listid;
+		YSI_g_sRequestModelSelection[playerid] = cb;
+	}
+	return ret;
+}
+
+stock ShowModelSelectionInlineEx(Func:cb<ii>, playerid, const itemsArray[], itemAmount, const headerText[], extraid, Float:Xrot = 0.0, Float:Yrot = 0.0, Float:Zrot = 0.0, Float:mZoom = 1.0, dialogBGcolor = 0x4A5A6BBB, previewBGcolor = 0x88888899, selectionColour = 0xFFFF00AA) 
+{
+	new ret = ShowModelSelectionMenuEx(playerid, itemsArray, itemAmount, headerText, extraid, Xrot, Yrot, Zrot, mZoom, dialogBGcolor, previewBGcolor, selectionColour);
+
+	if (ret) 
+	{
+		Indirect_Claim(cb);
+		YSI_g_PlayerCurrentListID[playerid] = extraid;
+		YSI_g_sRequestModelSelection[playerid] = cb;
+	}
+	return ret;
+}
+
+HOOK__ OnPlayerModelSelection(playerid, response, listid, modelid) 
+{
+	new 
+		Func:cb<ii> = YSI_g_sRequestModelSelection[playerid];
+
+	YSI_g_sRequestModelSelection[playerid] = Func:0<ii>;
+
+	if (listid != mS_INVALID_LISTID && listid == YSI_g_PlayerCurrentListID[playerid]) 
+	{
+		YSI_g_PlayerCurrentListID[playerid] = mS_INVALID_LISTID;
+		@.cb(response, modelid);
```

Comment:
Maybe try and see how y_dialog deals with parameters.  It can take 3 or 5, it would be good if this did the same - accepting either 2 or 4.  This is especially important when the callbacks aren't inlines, but publics, and you want `playerid` passing.